### PR TITLE
Week1 PR for Airport Challenge - With Weather Class

### PR DESCRIPTION
**Pull request by ViolaCrellin**
This nearly killed me.

### CODE REVIEW

File: spec/airport_spec.rb [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      end
+    end
+
+    context "Capacity constraints" do
+
+      describe "#full?" do
+
+      before do
+        plane1 = double("plane1", :flying? => true, :change_status => "Landed")
+        plane2 = double("plane2", :flying? => true, :change_status => "Landed")
```
Comment (ViolaCrellin) [inline spec/airport_spec.rb]: Rspec uses different syntax. Get with it hound.

File: spec/airport_spec.rb [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        expect(airport.land plane ).to eq "Plane has already landed"
+      end
+    end
+
+    context "Capacity constraints" do
+
+      describe "#full?" do
+
+      before do
+        plane1 = double("plane1", :flying? => true, :change_status => "Landed")
```
Comment (ViolaCrellin) [inline spec/airport_spec.rb]: Rspec uses different syntax - get with it hound!

File: spec/airport_spec.rb [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            allow(plane).to receive(:flying?).and_return(true)
+            airport2.land(plane)
+          end
+
+          it "Will not allow plane to take off if not sunny" do
+            allow(airport2).to receive(:empty?).and_return(false)
+            expect(airport2.take_off).to eq("Unsafe to take off plane whilst stormy")
+          end
+
+          it "Will not take off a plane from the holding bay whilst not sunny" do
```
Comment (ViolaCrellin) [inline spec/airport_spec.rb]: Needed full line.

File: spec/airport_spec.rb [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+          before do
+            allow(plane).to receive(:change_status)
+            allow(plane).to receive(:flying?).and_return(true)
+            airport2.land(plane)
+          end
+
+          it "Will not allow plane to take off if not sunny" do
+            allow(airport2).to receive(:empty?).and_return(false)
+            expect(airport2.take_off).to eq("Unsafe to take off plane whilst stormy")
```
Comment (ViolaCrellin) [inline spec/airport_spec.rb]: Needed full line.

File: lib/airport.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    @holding_bay = []
+    @capacity = capacity
+    @sunny = weather
+  end
+
+  def land(plane)
+    if full?
+      "The airport is full"
+    else
+      if @sunny && plane.flying?
```
Comment (ViolaCrellin) [inline lib/airport.rb]: Nope.

File: spec/plane_spec.rb [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+describe Plane do
+  subject(:plane) {described_class.new}
+  let(:airport) {double :airport}
+
+  describe "#initialize" do
+
+    context "All planes have a status, flying or landed" do
+      it {is_expected.to respond_to(:status)}
+
+      it "Ensures that, for Airtraffic control, Plane.new is initially flying" do
```
Comment (ViolaCrellin) [inline spec/plane_spec.rb]: Needed description.

File: lib/airport.rb [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    if empty?
+      "The airport is empty"
+    else
+        plane = @holding_bay.pop
+        if @sunny && plane.landed?
+          plane.change_status
+          plane
+        else
+          @holding_bay << plane
+          @sunny == false ? "Unsafe to take off plane whilst stormy" : "This plane is already flying"
```
Comment (ViolaCrellin) [inline lib/airport.rb]: Needed strings

**Comment by mechacoach** [comment]
You had your airport_challenge reviewed by **Edward Withers**

### The good points are:

* `context` or `describe` blocks are used (where appropriate) to group tests
* `before` blocks are used where appropriate
* There are a minimal number of untested paths in the code
* Unit tests all have single expectations
* Random behaviour is stubbed where appropriate
* The code follows Ruby naming conventions, Readability is supported by consistent naming
* Booleans are returned implicitly where appropriate, removeallcommented-outcode: There is no commented-out code in the submission
* Code avoids unnecessary public methods, Code avoids public access to internal objects (e.g. planes array)
* Classes have single responsibilities
* Code avoids strings where it should use symbols
* Command/Query naming is not an issue in the code
* The solution does not have any redundant code

### You should consider the following improvements:

* README should be updated
* Installation Instructions should be in the README
* Use named subject with `described_class`
* Avoid Vacuous tests
* Get rid of redundant `respond_to` expectations
* Remove all commented-out code
* Avoid magic numbers (e.g. on capacity)
* Improper use of attr_*

### Additional comments:
Awesome work.
Things to focus on:
- design of code for others to read, maintain, and extend
- refactoring methods to follow SRP, avoidmagicnumberse.g.oncapacity: Code avoids magic numbers

see https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md for more details

**Comment by mechacoach** [comment]
You had your airport_challenge reviewed by **Edward Withers**

### The good points are:

* `context` or `describe` blocks are used (where appropriate) to group tests
* `before` blocks are used where appropriate
* There are a minimal number of untested paths in the code
* Unit tests all have single expectations
* Random behaviour is stubbed where appropriate
* The code follows Ruby naming conventions, Readability is supported by consistent naming
* Booleans are returned implicitly where appropriate, removeallcommented-outcode: There is no commented-out code in the submission
* Code avoids unnecessary public methods, Code avoids public access to internal objects (e.g. planes array)
* Classes have single responsibilities
* Code avoids strings where it should use symbols
* Command/Query naming is not an issue in the code
* The solution does not have any redundant code

### You should consider the following improvements:

* README should be updated
* Installation Instructions should be in the README
* Use named subject with `described_class`
* Avoid Vacuous tests
* Get rid of redundant `respond_to` expectations
* Remove all commented-out code
* Avoid magic numbers (e.g. on capacity)
* Improper use of attr_*

### Additional comments:
Awesome work.
Things to focus on:
- design of code for others to read, maintain, and extend
- refactoring methods to follow SRP, avoidmagicnumberse.g.oncapacity: Code avoids magic numbers

see https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md for more details

**Pull request closed without merging**